### ISSUE TITLE
Add disable caching option to the compiler

### ISSUE THREAD

**Issue by koachan** [ziglang/zig#7441]
When working with the compiler, it's often necessary to delete all the cached outputs before debugging to make the compiler redo everything from the start.
Deleting it manually can be a tedious process, so it'd be nice if there's a compiler option that would disable all caching.

**Comment by ifreund** [comment]
A shell alias could make this painless:

```
alias zig="rm -r ~/.cache/zig; rm -r zig-cache; zig"
```

**Comment by koachan** [comment]
But it doesn't remove $ZIGSRCDIR/test/zig-cache or other random places where it might place its caches

**Comment by ifreund** [comment]
> But it doesn't remove $ZIGSRCDIR/test/zig-cache or other random places where it might place its caches

I agree, it's not perfect and a compiler option would certainly be more developer friendly.

**Comment by andrewrk** [comment]
Compiler has a hard dependency on the file system. If you want to ignore caching, give it a temporary directory and delete it after invoking the compiler.
